okhttp/src/main/java/okhttp3/CipherSuite.java

```diff
@@ -29,356 +32,421 @@
  * <p>See also <a href="https://android.googlesource.com/platform/external/conscrypt/+/master/src/main/java/org/conscrypt/NativeCrypto.java">NativeCrypto.java</a>
  * from conscrypt, which lists the cipher suites supported by Android.
  */
-public enum CipherSuite {
+public final class CipherSuite {
+  // for interning - must be at top of file
+  private static final ConcurrentMap<String, CipherSuite> INSTANCES = new ConcurrentHashMap<>();
```

Comment:
Reverted back to ConcurrentMap because WeakHashMap doesn't work when the value objects have a strong reference to the keys.